crates/tokenlist/src/lib.rs

```diff
@@ -0,0 +1,59 @@
+use serde::{ Deserialize, Serialize };
+use reqwest::Error;
+use chrono::{ DateTime, Utc };
+use std::{
+    fs::File,
+    io::Write,
+};
+
+#[derive(Debug, Deserialize, Serialize )]
+pub struct TokenList {
+    updated_at: Option<DateTime<Utc>>,
+    name: String,
+    tokens: Vec<Token>,
+}
+
+#[derive(Deserialize, Debug, Serialize)]
+pub struct Token {
+    #[serde(rename = "chainId")]
+    chain_id: i32,
+    
+    address: String,
+    name: String,
+    //symbol: String,
+    //decimals: i32,
+
+    #[serde(rename = "logoURI")]
+    logo_uri: String,
+}
+
+pub async fn init() -> Result<(), Error> {
+
+    let response = match reqwest::get("https://gateway.ipfs.io/ipns/tokens.uniswap.org").await {
```

Comment:
the idea of the local timestamp is to avoid doing this request if the timestamp is too recent. so the flow should actually be to read the current timestamp first, if any